### ISSUE TITLE
js compressor can't handle non-ASCII characters inside inline Javascript

### ISSUE THREAD

**Issue by pbgc** [mintchaos/django_compressor#11]
Don't know if this happens also with non-ASCII caracters inside linked js files but if an inline Javascript block contains non-ASCII caracters, for example a Javascript String, like its usual for non-US users like me the following error occurs:

```
TemplateSyntaxError at /accounts/login/
Caught an exception while rendering: 'ascii' codec can't encode characters in position 429-430: ordinal not in range(128)

Original Traceback (most recent call last):
  File "/home/pbeck/django_virtualenv/lib/python2.5/site-packages/django/template/debug.py", line 71, in render_node
    result = node.render(context)
  File "/home/pbeck/dev_projects/django-projects/siva/../siva/apps_3p/compressor/templatetags/compress.py", line 22, in render
    in_cache = cache.get(compressor.cachekey)
  File "/home/pbeck/dev_projects/django-projects/siva/../siva/apps_3p/compressor/__init__.py", line 65, in cachekey
    return "django_compressor.%s" % get_hexdigest(cachestr)[:12]
  File "/home/pbeck/dev_projects/django-projects/siva/../siva/apps_3p/compressor/__init__.py", line 22, in get_hexdigest
    return hashlib.sha1(plaintext).hexdigest()
UnicodeEncodeError: 'ascii' codec can't encode characters in position 429-430: ordinal not in range(128)
```

**Comment by pbgc** [comment]
After a quick try to fix this, at least for me and for what I need the following solves my problem:

in **init**.py changing:

@property
    def cachekey(self):
        cachebits = [self.content]
        cachebits.extend([str(m) for m in self.mtimes])
        cachestr = "".join(cachebits)
        return "django_compressor.%s" % get_hexdigest(cachestr)[:12]

to

@property
    def cachekey(self):
        cachebits = [self.content]
        cachebits.extend([str(m) for m in self.mtimes])
        cachestr = "".join(cachebits)
        return "django_compressor.%s" % get_hexdigest(cachestr.encode("utf-8"))[:12]

**Comment by muhuk** [comment]
[This patch](http://github.com/muhuk/django_compressor/commit/e3e8010bb642a1fee6740dfa516cb451813c6a03) should fix. (with tests).

Please note the comment at compressor.Compressor.concat()

**Comment by camilonova** [comment]
works like charm, Thanks.

**Comment by jezdez** [comment]
Fixed in 768fb42241ac90a6239472873a46ef087b69b328.
